zrml/prediction-markets/src/lib.rs

```diff
@@ -907,12 +908,16 @@ mod pallet {
         pub fn reject_market(
             origin: OriginFor<T>,
             #[pallet::compact] market_id: MarketIdOf<T>,
+            reject_reason: Vec<u8>,
         ) -> DispatchResultWithPostInfo {
             T::RejectOrigin::ensure_origin(origin)?;
             let market = T::MarketCommons::market(&market_id)?;
             let open_ids_len = Self::clear_auto_open(&market_id)?;
             let close_ids_len = Self::clear_auto_close(&market_id)?;
-            Self::do_reject_market(&market_id, market)?;
+            let reject_reason = reject_reason
+                .try_into()
```

Comment:
isn't that O(1). As per following because Vec's len() is O(1).
```rust
  27 impl<T, S: Get<u32>> TryFrom<Vec<T>> for BoundedVec<T, S> {                                                                                  
  26     type Error = Vec<T>;                                                                                                                     
  25     fn try_from(t: Vec<T>) -> Result<Self, Self::Error> {                                                                                    
  24         if t.len() <= Self::bound() {                                                                                                        
  23             // explicit check just above                                                                                                     
  22             Ok(Self::unchecked_from(t))                                                                                                      
  21         } else {                                                                                                                             
  20             Err(t)                                                                                                                           
  19         }                                                                                                                                    
  18     }                                                                                                                                        
  17 } 
```